src/openai/dto/apiRequest.dto.ts

```diff
@@ -1,5 +1,32 @@
+// if you think it is better to seperate the interfaces into different entity files, please let me know.
+export interface Message {
+    name: string;
+    content: string;
+    role: Role;
+}
+
+export interface Chat {
+    chatId: number;
+    customerId: number;
+    messages: Message[];
+}
+
+export interface User {
+    userId: number;
+    username: string;
+    roles?: Role[];
+}
+
+enum Role {
+    ASSISTANT = 'assistant',
+    USER = 'user',
+    SYSTEM = 'system',
+    FUNCTION = 'function',
+}
+
 export class apiRequestDTO {
-    _id: string;
-    chatId: string;
-    chatConversation: Record<string, any>;
+    chatId: number;
+    customerId: number;
+    user: User;
```

Comment:
you also dont need the user Id and the user, the only thing that you need is the messages array

Reply:
I do need the User object, so i can extract the userName and Role, and put it inside each Message object that is send to the API.

Reply:
the fact that you send the username to the api will give you no advantage at all, if anything, it will only make your code less scaleable. the username wont improve you guide nor make it worse, it wont have any effect for your guides if you include or exclude the username